### PR TITLE
Wicked Witch Scan: Add new website

### LINKED ISSUE #129: Wicked Witch Scan changed URL
### Source information

Wicked Witch Scan 1.4.32

### Source language

Português (Brasil)

### Source new URL

https://wicked-witch-scan.com/

### Other details

_No response_

### Acknowledgements

- [X] I have updated all installed extensions.
- [X] I have opened WebView and checked that the source URL is not updated yet.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by beer-psi**
Closes #129 

~~Since their old website is still active and not all content has been migrated over, Wicked Witch Scan is now a source factory containing "Wicked Witch Scan" and "Wicked Witch Scan (Novo)". The old source will be removed when the team shuts down the old website.~~

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

**Comment by vetleledaal** [comment]
Not sure about appending "(Novo)", would cause issues next time they switch sites. Renaming the old one isn't great either, as downloads would have to be moved.

**Comment by beer-psi** [comment]
> Not sure about appending "(Novo)", would cause issues next time they switch sites. Renaming the old one isn't great either, as downloads would have to be moved.

Need something to differentiate the sources on the user side temporarily though. Would making the displayed name "Wicked Witch Scan (Novo)", but the name used for ID generation just "Wicked Witch Scan" (with `versionId = 2` of course) work?

**Comment by AwkwardPeak7** [comment]
I think you should just add their new site as a new separate extension considering it doesn't use Madara. Old one can be removed later

**Comment by beer-psi** [comment]
> I think you should just add their new site as a new separate extension considering it doesn't use Madara. Old one can be removed later

What would the naming be for the new package then? Icon is probably not a problem because the old Madara website used the default icon. If we used `pt.wickedwitchscannew`, that'd cause issues if they switch sites again, like Vetle said.

**Comment by AwkwardPeak7** [comment]
pt.wickedwitchscansnovo?  Package name can be kept if they change site again

**Comment by beer-psi** [comment]
> pt.wickedwitchscansnovo? Package name can be kept if they change site again

There might be a chance of something like this (i.e. old and new coexisting) happening again though. 

> I think you should just add their new site as a new separate extension considering it doesn't use Madara.

It's stuck in Madara because of the old site. When the old site goes offline, this will become an individual extension.

**Comment by beer-psi** [comment]
There's still the issue of the downloads folder conflicting though. Not sure how to move forward.

**Comment by beer-psi** [comment]
I think I'm moving forward with splitting the new website into an individual extension (`pt.wickedwitchscannovo`). If something like this happens again we can probably recycle `pt.wickedwitchscan` by then.

File: src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+import java.util.concurrent.TimeUnit
+import java.util.zip.ZipInputStream
+
+@SuppressLint("WrongConstant")
+class WickedWitchScan : ParsedHttpSource() {
+
+    override val name = "Wicked Witch Scan"
```
Comment (AwkwardPeak7) [inline src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt:48]: change the name here as id should never be same for two sources

File: src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+            addPathSegments("auto-complete/")
+            addQueryParameter("term", query)
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val manga = json.parseToJsonElement(response.body.string()).jsonArray.map {
+            val element = Jsoup.parse(it.jsonObject["html"]!!.jsonPrimitive.content)
```
Comment (AwkwardPeak7) [inline src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt]:
```suggestion
            val element = Jsoup.parseBodyFragment(it.jsonObject["html"]!!.jsonPrimitive.content)
```

File: src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+        val images = generateSequence { zis.nextEntry }
+            .map {
+                val entryName = it.name
+                val splitEntryName = entryName.split('.')
+                val entryIndex = splitEntryName.first().toInt()
+                val entryType = splitEntryName.last()
+
+                val imageData = if (entryType == "avif") {
+                    Log.d("wickedwitch", "Extracting AVIF image $entryName")
```
Comment (AwkwardPeak7) [inline src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt]: remove logs

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt (hunk 1 above)
Comment (beer-psi) [reply src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt:48]: I already bumped the versionId though?

```kotlin
    // Source changed from Madara to homegrown website
    override val versionId = 2
```

Comment (AwkwardPeak7) [reply src/pt/wickedwitchscannovo/src/eu/kanade/tachiyomi/extension/pt/wickedwitchscannovo/WickedWitchScan.kt:48]: Oh right nvm then 🫠

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**